src/utils/postgres-client/index.ts

```diff
@@ -0,0 +1,379 @@
+    const {
+      rows: [user],
+    } = await client.query<SqlUser>(createUserQueryByColumn('email'), [email]);
+    client.release();
+    return cameliseUser(user);
+  },
+
+  getUserByPhoneNumber: async (phoneNumber: string) => {
+    const client = await pool.connect();
+    const {
+      rows: [user],
+    } = await client.query<SqlUser>(createUserQueryByColumn('phone_number'), [
+      phoneNumber,
+    ]);
+    client.release();
+    return cameliseUser(user);
+  },
+
+  getUserByPhoneNumberAndOtp: async (phoneNumber: string) => {
+    const client = await pool.connect();
+    const {
+      rows: [user],
+    } = await client.query<SqlUser>(
+      createUserQueryWhere(
+        `u.phone_number = $1 AND u.otp_method_last_used = 'sms' AND u.otp_hash_expires_at > NOW()`
+      ),
+      [phoneNumber]
+    );
+    client.release();
+    return cameliseUser(user);
+  },
+
+  getUserByTicket: async (ticket: string) => {
+    const client = await pool.connect();
+    const {
+      rows: [user],
+    } = await client.query<SqlUser>(
+      createUserQueryWhere(`u.ticket = $1 and u.ticket_expires_at > NOW()`),
+      [ticket]
+    );
+    client.release();
+    return cameliseUser(user);
+  },
+
+  getUserByChallence: async (challenge: string) => {
+    const client = await pool.connect();
+    const {
+      rows: [user],
+    } = await client.query<SqlUser>(
+      createUserQueryByColumn('webauthn_current_challenge'),
+      [challenge]
+    );
+    client.release();
+    return cameliseUser(user);
+  },
+
+  getUserByProvider: async (providerId: string, providerUserId: string) => {
+    const client = await pool.connect();
+    const { rows } = await client.query<{ user_id?: string; id: string }>(
+      `SELECT user_id, id FROM "auth"."user_providers" WHERE up.provider_id = $1 AND up.provider_user_id = $2;`,
+      [providerId, providerUserId]
+    );
+    let user: SqlUser | null = null;
+    const userId = rows[0]?.user_id;
+    if (userId) {
+      user = await getUserById(client, userId);
+    }
+    client.release();
+    return { id: rows[0].id, user: cameliseUser(user) };
+  },
+
+  updateAuthUserprovider: async (
+    id: string,
+    {
+      accessToken,
+      refreshToken,
+    }: { accessToken?: string; refreshToken?: string }
+  ) => {
+    const client = await pool.connect();
+    await client.query(
+      `UPDATE "auth"."user_providers" SET access_token = $1, refresh_token = $2 WHERE id = $3;`,
+      [accessToken, refreshToken, id]
+    );
+    client.release();
+  },
+
+  insertUser: async (user: Partial<User>) => {
+    const client = await pool.connect();
+    const transformedUser = snakeiseUser(user) as Partial<SqlUser>;
+    const { roles, ...rest } = transformedUser;
+    const columns = Object.keys(rest);
+    const values = Object.values(rest);
+    const {
+      rows: [insertedUser],
+    } = await client.query<SqlUser>(
+      `INSERT INTO "auth"."users" (${columns.join(',')}) VALUES(${[
```

Comment:
this is necessary because the user columns being filled will depend on the sign-up method used, correct?

Reply:
Exactly